### DISCUSSION TITLE
Support for SolidJS

### DISCUSSION THREAD

**Discussion by thedanchez** [fireproof-storage/fireproof#42] [Ideas]
Just wanted to plant this seed as food for thought and to gauge interest. I feel like it would be cool to have a SolidJS adapter for Fireproof considering how close it resembles React. It's also a much more performant piece of technology by comparison across many metrics.

I may have time down the road to maybe try doing it myself (can't make any promises on my free time though), but was curious if it was something that was on the radar? Or would something like that ever be considered for future possibilities?

Moreover, if someone were interested in tackling this problem, any ideas how big of a lift it would be? Is it simply a matter of following the `react` package that currently exists?

**Comment by jchris** [comment]
To get started, you would probably just start using Fireproof in your SolidJS app via `@fireproof/core`. When you do this in a React app, you end up with a lot of `useEffect` and `setState` calls, which are what ended up hidden inside `useLiveQuery` etc.

For SolidJS and other frameworks the same story should hold. The core of the React hooks is a listener for `db.subscribe()` callbacks.
